### PR TITLE
Add validation for age for age equal to 0

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://www.pivotaltracker.com/story/show/170501005

### CODE REVIEW

**Comment by sanchita** [comment]
Please fix code climate issue.

**Comment by janhavisinghh** [comment]
> Please fix code climate issue.

Done

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt [hunk 1]
```diff
@@ -350,12 +352,20 @@ class EditPatientScreen(context: Context, attributeSet: AttributeSet) : Relative
           showDateOfBirthIsInFutureError()
         }
 
-        AGE_INVALID -> {
-          showAgeIsInvalidError()
+        AGE_EXCEEDS_MAX_LIMIT -> {
+          showAgeExceedsMaxLimitError()
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt]: If its just one statement in `when` block, you can get rid of the braces.
Please format this method to remove braces everywhere.

File: app/src/main/java/org/simple/clinic/widgets/ageanddateofbirth/UserInputAgeValidator.kt [hunk 2]
```diff
@@ -14,21 +15,26 @@ class UserInputAgeValidator @Inject constructor(
 ) {
   sealed class Result {
     object Valid : Result()
-    object Invalid : Result()
+    sealed class Invalid : Result() {
+      object ExceedsMaxAge : Invalid()
+      object ExceedsMinAge : Invalid()
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/widgets/ageanddateofbirth/UserInputAgeValidator.kt]: Can you rename to `ExceedsMaxAgeLimit` and `ExceedsMinAgeLimit`?

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -82,6 +82,7 @@
   <string name="patiententry_consent_second_para">Anyone authorized by you, or the healthcare provider for whom you work, will have access to the patient data and information you enter here. By using the app you confirm that the patient understands who will have access to the data obtained through this app and consents to that access.</string>
   <string name="patiententry_consent_whatsapp_sms_reminders">WhatsApp / SMS Reminders</string>
   <string name="patiententry_invalid_age_error">Age cannot be more than 120 years</string>
+  <string name="patiententry_age_exceeds_min_limit_error">Age cannot be equal to 0 years</string>
```
Comment (sanchita) [inline app/src/main/res/values/strings.xml]: Did we get this text verified with someone from the design team?

File: app/src/test/java/org/simple/clinic/newentry/UserInputAgeValidatorTest.kt [hunk 4]
```diff
@@ -95,4 +96,16 @@ class UserInputAgeValidatorTest {
     //then
     assertThat(validation).isEqualTo(Valid)
   }
+
+  @Test
+  fun `when user inputs age equal to 0 years, then return invalid error`() {
+    //given
+    val age = 0
```
Comment (sanchita) [inline app/src/test/java/org/simple/clinic/newentry/UserInputAgeValidatorTest.kt]: Please add a test for date of birth as well.

File: app/src/main/java/org/simple/clinic/newentry/PatientEntryScreen.kt [hunk 5]
```diff
@@ -442,20 +452,35 @@ class PatientEntryScreen(context: Context, attrs: AttributeSet) : RelativeLayout
...
+    ageEditTextInputLayout.error = when {
+      show -> resources.getString(R.string.patiententry_invalid_age_error)
+      else -> null
     }
+  }
 
-  override fun showInvalidDateOfBirthAgeError(show: Boolean) {
+  override fun showDOBExceedsMaxLimitError(show: Boolean) {
     dateOfBirthInputLayout.error = when {
       show -> resources.getString(R.string.patiententry_invalid_age_error)
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/newentry/PatientEntryScreen.kt]: Rename the string resource to be specific.

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt [hunk 6]
```diff
@@ -315,58 +315,20 @@ class EditPatientScreen(context: Context, attributeSet: AttributeSet) : Relative
...
+        COLONY_OR_VILLAGE_EMPTY -> showEmptyColonyOrVillageError(true)
+        DISTRICT_EMPTY -> showEmptyDistrictError(true)
+        STATE_EMPTY -> showEmptyStateError(true)
+        BOTH_DATEOFBIRTH_AND_AGE_ABSENT -> showAgeEmptyError(true)
+        DATE_OF_BIRTH_PARSE_ERROR -> showInvalidaDateOfBithError()
+        DATE_OF_BIRTH_IN_FUTURE -> showDateOfBirthIsInFutureError()
+        AGE_EXCEEDS_MAX_LIMIT -> showAgeExceedsMaxLimitError()
+        DATE_OF_BIRTH_EXCEEDS_MAX_LIMIT -> showDateOfBirthExceedsMaxLimitError()
+        AGE_EXCEEDS_MIN_LIMIT -> showAgeExceedsMinLimitError()
+        DATE_OF_BIRTH_EXCEEDS_MIN_LIMIT -> showDateOfBirthExceedsMinLimitError()
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt]: Awesome!

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt [hunk 7]
```diff
@@ -315,58 +315,20 @@ class EditPatientScreen(context: Context, attributeSet: AttributeSet) : Relative
   override fun showValidationErrors(errors: Set<EditPatientValidationError>) {
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt]: Rename commit message to `Reformat EditPatientScreen#showValidationErrors`

**Comment by sanchita** [comment]
Left a few comments. PTAL.
Also rebase your branch with master.

File: app/src/main/java/org/simple/clinic/widgets/ageanddateofbirth/UserInputAgeValidator.kt (hunk 2 above)
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/widgets/ageanddateofbirth/UserInputAgeValidator.kt]: Done

File: app/src/main/java/org/simple/clinic/newentry/PatientEntryScreen.kt (hunk 5 above)
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/newentry/PatientEntryScreen.kt]: Done

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (janhavisinghh) [reply app/src/main/res/values/strings.xml]: Yes, Varenya confirmed this.

File: app/src/test/java/org/simple/clinic/newentry/UserInputAgeValidatorTest.kt (hunk 4 above)
Comment (janhavisinghh) [reply app/src/test/java/org/simple/clinic/newentry/UserInputAgeValidatorTest.kt]: Done

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt (hunk 7 above)
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt]: Done

**Comment by janhavisinghh** [comment]
> Left a few comments. PTAL.
> Also rebase your branch with master.

Done all the changes requested

**Pull request merged**